src/components/TodoForm/TodoForm.tsx

```diff
@@ -0,0 +1,78 @@
+import React, { useState } from 'react';
+import cN from 'classnames';
+import { Todo } from '../../types/Todo';
+import { TodoPatch } from '../../types/TodoPatch';
+
+type Props = {
+  todos: Todo[],
+  onAddTodo: (title: string) => void,
+  onUpdateTodo: (todoId: number, data: TodoPatch) => void,
+};
+
+export const TodoForm: React.FC<Props> = ({
+  todos,
+  onAddTodo,
+  onUpdateTodo,
+}) => {
+  const [todoTitle, setTodoTitle] = useState('');
+  const [isLoading, setIsLoading] = useState(false);
+
+  const hasTodos = Boolean(todos.length);
+  const hasActiveTodos = todos.filter(todo => !todo.completed);
+  const hasAllCompleted = todos.every(todo => todo.completed);
+
+  const handleSubmit = async (event: React.FormEvent) => {
+    event.preventDefault();
+
+    setIsLoading(true);
+
+    await onAddTodo(todoTitle);
```

Comment:
Asynchronous requests should be placed in the middle of a `try-catch` block to catch errors.

Reply:
@roman-mirzoian I have a try-catch construction inside this function at App.tsx